pymatgen/core/structure.py

```diff
@@ -625,6 +625,35 @@ def add_spin_by_site(self, spins: Sequence[float]) -> None:
                 new_species[Species(sym, oxidation_state=oxi_state, spin=spin)] = occu
             site.species = Composition(new_species)
 
+    def add_interstitial(self, atom: str) -> None:
+        """Creates an interstitial defect in the structure in-place by adding the atom to the available voids.
+
+        Args:
+            atom (str): e.g. "H", "Li" etc.
+        """
+
+        def density(r, R):
+            return np.exp(-np.dot(r - R, r - R))
+
+        supercell = self.copy()
+        supercell.make_supercell([3, 3, 3])
+        divisions = np.array([10, 10, 10]) * 3
```

Comment:
It might be worth thinking about if we can generalize this in any way. I assume the 3x3x3 was to address PBCs? If so, it could make more sense to test for the minimum image distance rather than make a supercell, particularly if the starting structure is large to begin with.